### DISCUSSION TITLE
Possible to have different projects with different vcpkg.json and share a common package folder (x-packages-root)

### DISCUSSION THREAD

**Discussion by runarundheim** [microsoft/vcpkg#34410] [Ideas]
I would be nice if it was possible to have a common package folder for all projects in the build job. In the attached zip file there are a solution with two projects with (vcpkg packages):

- Main (icu, zlib)
- ProjectSub1 (icu, openssl)

Microsoft Visual Studio Version 17.8.0 Preview 3.0 (with 'vcpkg package manager' installed) is used to build it.

The project is set up to share the x-package-root. When the main is build it will delete icu content in package folder and copy it again. The problem with it are that the hardlink created will then be normal files. So it will take more disk space. The packages used in this example are small, but if it changes to use qt then they will be very big.

I would prefer if it is possible to use the x-packages-root or another argument to set a common package folder. Then it could be possible to have hardlink for the files used in the x-install-root and that folder could be private for the project.

[Vcpkg_test.zip](https://github.com/microsoft/vcpkg/files/12867413/Vcpkg_test.zip)

**Comment by dg0yt** [comment]
The packages dir is just a staging area, and helpful in case of package build errors. If you are concerned about disk space, use `--clean-packages-after-build`.

**Reply by runarundheim** [reply]
That will not help since the install directory only is a hard link to the package directory. So the clean will not clean up much.

The problem are that if you have 10 sub projects and all projects have their own vcpkg.json file then you have 10 copies of lib, include, dll, and pch files. This take up a lot of data. If the sub projects could share the package directory you would only get 1 copy and a lot of hard links.

**Reply by dg0yt** [reply]
There is also `--clean-buildtrees-after-build` - the buildtrees are never hardlinked, and they can take a lot of space - unless binary artifacts are installed from cache.

**Comment by dg0yt** [comment]
If you want to share an installation root, use classic mode, not manifest mode. 
A single installation root can represent only a single manifest. Every different manifest demands a different set of files.
